src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -409,24 +409,28 @@ def on_request(self, request):
 
         # 'request_data' has been dumped into JSON string in set_json_body() when building HttpRequest in Python SDK.
         # In order to facilitate subsequent parsing, it is converted into a dict first
-        http_request.data = json.loads(request_data)
+        modified_data = json.loads(request_data)
 
-        if http_request.data.get('properties', {}).get('template'):
-            template = http_request.data["properties"]["template"]
-            del http_request.data["properties"]["template"]
+        if modified_data.get('properties', {}).get('template'):
+            template = modified_data["properties"]["template"]
+            del modified_data["properties"]["template"]
 
             # templateLink and template cannot exist at the same time in deployment_dry_run mode
-            if "templateLink" in http_request.data["properties"].keys():
-                del http_request.data["properties"]["templateLink"]
+            if "templateLink" in modified_data["properties"].keys():
+                del modified_data["properties"]["templateLink"]
 
             # The 'template' and other properties (such as 'parameters','mode'...) are spliced and encoded into the UTF-8 bytes as the request data
             # The format of the request data is: {"properties": {"parameters": {...}, "mode": "Incremental", template:{\r\n  "$schema": "...",\r\n  "contentVersion": "...",\r\n  "parameters": {...}}}
             # This is not an ordinary JSON format, but it is a JSONC format that service can deserialize
             # If not do this splicing, the request data generated by default serialization cannot be deserialized on the service side.
             # Because the service cannot deserialize the template element: "template": "{\r\n  \"$schema\": \"...\",\r\n  \"contentVersion\": \"...\",\r\n  \"parameters\": {...}}"
-            partial_request = json.dumps(http_request.data)
+            partial_request = json.dumps(modified_data)
             http_request.data = partial_request[:-2] + ", template:" + template + r"}}"
-            http_request.data = http_request.data.encode('utf-8')
```

Comment:
I removed the encoding from string -> bytes. Conversion from one to the other feels unexpected, because this pipeline is explicitly only operating on a string value (bytes are ignored on line 407).